server/data_common/matrix_loader.py

```diff
@@ -97,13 +102,19 @@ def __init__(self):
     @contextmanager
     def data_adaptor(self, location, app_config):
         # create a loader for to this location if it does not already exist
+
+        delete_adaptor = None
+        data_adaptor = None
+
         with self.lock:
             value = self.datasets.get(location)
             if value is not None:
                 cache_item = value[0]
                 last_accessed = time.time()
                 self.datasets[location] = (cache_item, last_accessed)
-            else:
+                data_adaptor = cache_item.acquire_existing()
+
+            if data_adaptor is None:
                 while True:
                     # find the last access times for each loader
                     items = list(self.datasets.items())
```

Comment:
for performance, I would move the sorted() to below the test for `< self.MAX_CACHED`.  No reason to sort if you aren't going to eject something

Reply:
Done